README.md

```diff
@@ -129,6 +129,10 @@ These GitHub repositories provide supplementary resources for Rush Stack:
 | [/build-tests-samples/heft-web-rig-library-tutorial](./build-tests-samples/heft-web-rig-library-tutorial/) | (Copy of sample project) Building this project is a regression test for Heft |
```

Comment:
![image](https://github.com/microsoft/rushstack/assets/4673363/2a95fcf8-760c-415e-af2a-3f5e12840a02)

***It can be fixed in a separate PR,*** but this log notice can be improved.  

1. Like the build cache notices, it should only be shown when invoking `rush build --verbose`.
2. Also I would suggest to improve the formatting of the message:

   ```
   pnpm-sync: Copied 107 files in 95ms from C:\Git\rushstack\libraries\terminal
   ```
   
3. It seems `pnpmSyncCopy()` is printing this message [directly to the console](https://github.com/tiktok/pnpm-sync/blob/ffd80ce4e5c1eea7b904325f79f11dc17148fd1a/packages/pnpm-sync-lib/src/pnpmSyncCopy.ts#L65).  We should improve the API to provide a messaging callback that allows the output to be redirected and (ideally) optionally reformatted.  Something like this:

   ```ts
   pnpmSyncCopy({
     pnpmSyncJsonPath,
     messageCallback: ({ message, messageKind: 'error'|'warning'|'info'|'verbose'|'timing', messageId, details }) => {
       if (messageKind === 'verbose' && !debug) {
         return;
       }
       switch (messageId) {
         case 'sync-finished':
           // customized logging; the structure of details can depend on messageId
           terminal.writeLine(colors.green('pnpm-sync') 
             + ` copied ${details.fileCount} files in ${details.totalMs} ms from ${details.sourcePath}`);
           break;
         default:
           // simple preformatted logging
           if (messageKind === 'error' | messageKind === 'warning') {
             console.error(message);
           } else {
             console.log(message);
           }
           break;
       }
     }
   })
   ```
   This would ensure that Rush (and in the future PNPM) has full ownership of its CLI UX.
   
@g-chao @iclanton

Reply:
https://github.com/tiktok/pnpm-sync/issues/14